mephisto/abstractions/blueprints/static_html_task/source/dev/app.jsx

```diff
@@ -98,11 +98,23 @@ function MainApp() {
 
 function SubmitFrame({ children, onSubmit, currentTask }) {
   const [submitting, setSubmitting] = React.useState(false);
+  const [isHidden, setIsHidden] = React.useState(false);
+
+  const handler = React.useCallback(
+    (hideValue) => {
+      if (typeof hideValue == "boolean") setIsHidden(hideValue);
+    },
+    [setIsHidden]
+  );
 
   React.useEffect(() => {
     // Reset submitting when switching from onboarding
     setSubmitting(false);
-  }, [currentTask]);
+  }, []);
+
+  React.useEffect(() => {
+    window._MEPHISTO_CONFIG_.EVENT_EMITTER.on("HIDE_SUBMIT_BUTTON", handler);
+  }, [setIsHidden]);
```

Comment:
Runs the hideSubmitButton handler method when the HIDE_SUBMIT_BUTTON event is emitted.